### ISSUE TITLE
Bug: [no-useless-template-literals] Nested template literals contain useless one

### ISSUE THREAD

**Issue by waynzh** [typescript-eslint/typescript-eslint#8584]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.3&fileType=.ts&code=MYewdgzgLgBGCuBbGBeGBGA3AKAAYEMASAb1wCMSFEBfXa4XTIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tmUSWWS18iALbF4AQ2G1Gw6BPioMkRNGgdokcGAC%2BIHUA&tsconfig=&tokens=false

### Repro Code

```TypeScript
const num = 1;
`a${`b${num}`}c`;
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-useless-template-literals": "error"
  }
}
```

### Expected Result

Delete the useless template literal:

\`ab${num}c\`

### Actual Result

\`a${\`b${num}\`}c\`

### Additional Info

See https://github.com/typescript-eslint/typescript-eslint/pull/8575#discussion_r1507616393.
This [valid case](https://github.com/typescript-eslint/typescript-eslint/pull/8575/files#diff-e1397d81c6335b2044a2b15a62abf713bd821075d1a7b1d1f37521f031b1acefR66) in this PR needs to be adjusted if that is merged.
